display_information/control-annotation-sublayer-visibility/src/main/java/com/esri/samples/control_annotation_sublayer_visibility/ControlAnnotationSublayerVisibilitySample.java

```diff
@@ -0,0 +1,184 @@
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/style.css").toExternalForm());
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Control Annotation Sublayer Visibility");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create checkboxes for toggling the sublayer visibility manually
+      CheckBox closedSublayerCheckbox = new CheckBox();
+      closedSublayerCheckbox.setSelected(true);
+      CheckBox openSublayerCheckbox = new CheckBox();
+      openSublayerCheckbox.setSelected(true);
+
+      // create a control panel and label for the checkboxes
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.3)"), CornerRadii.EMPTY,
+              Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setMaxSize(180, 50);
+      controlsVBox.getStyleClass().add("panel-region");
+      Label controlsVBoxLabel = new Label("Annotation Sublayer Visibility");
+      // add checkboxes and label to the control panel
+      controlsVBox.getChildren().addAll(controlsVBoxLabel, closedSublayerCheckbox, openSublayerCheckbox);
+
+      // create a map view
+      mapView = new MapView();
+
+      // show current map scale in a label at the bottom of the screen
+      Label currentMapScaleLabel = new Label();
+      currentMapScaleLabel.setTextFill(Color.WHITE);
+      mapView.addMapScaleChangedListener(mapScaleChangedEvent -> currentMapScaleLabel.setText("Scale: 1:" + Math.round(mapView.getMapScale())));
+
+      // load the mobile map package
+      mobileMapPackage = new MobileMapPackage("./samples-data/mmpk/GasDeviceAnno.mmpk");
+      mobileMapPackage.loadAsync();
+      mobileMapPackage.addDoneLoadingListener(() -> {
+        if (mobileMapPackage.getLoadStatus() == LoadStatus.LOADED) {
+          // set the mobile map package's map to the map view
+          mapView.setMap(mobileMapPackage.getMaps().get(0));
+          // find the annotation layer withing the map
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            if (layer instanceof AnnotationLayer) {
+              // load the annotation layer
+              layer.loadAsync();
+              layer.addDoneLoadingListener(() -> {
+                if (layer.getLoadStatus() == LoadStatus.LOADED) {
+                  // get annotation sublayer name from sublayer content
+                  AnnotationSublayer closedSublayer = (AnnotationSublayer) layer.getSubLayerContents().get(0);
+                  AnnotationSublayer openSublayer = (AnnotationSublayer) layer.getSubLayerContents().get(1);
+
+                  // set the layer name for the checkboxes
+                  closedSublayerCheckbox.setText(buildLayerName(closedSublayer));
```

Comment:
Was there a reason this is abstracted out as a method? It hides away the .getName functionality and scale of the annotation layers, which we want to showcase. How about something simple like this instead?
```
 closedSublayerCheckbox.setText(closedSublayer.getName());
 openSublayerCheckbox.setText(openSublayer.getName() + 
        " (" + "1:" + Math.round(openSublayer.getMaxScale()) + 
        " - 1:" + Math.round(openSublayer.getMinScale()) + ")");
```

Reply:
LGTM. I was following the Android implementation, and couldn't resist a good method 😄 . Changed now